composer/algorithms/utils/augmentation_common.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from typing import Type, TypeVar, cast
+
+import torch
+from PIL.Image import Image as PillowImage
+from torchvision import transforms
+
+_InputImgT = TypeVar("_InputImgT", torch.Tensor, PillowImage)
+_OutputImgT = TypeVar("_OutputImgT", torch.Tensor, PillowImage)
+
+
+def image_as_type(image: _InputImgT, need_type: Type[_OutputImgT]) -> _OutputImgT:
+    """Creates a copy of an image-like object with a different type.
+
+    Args:
+        image: a tensor or PIL image. Batches of images in NCHW format are
+            also supported.
+        need_type: type of the copied image
+
+    Returns:
+        A copy of ``image`` with type ``need_type``.
+
+    Raises:
+        RuntimeError: if ``image`` is batch of images, but ``need_type`` is
+            a PIL image. We could convert to a batch of PIL images, but we
+            instead fail fast since this is not expected behavior at any
+            call sites (within data augmenation functions).
```

Comment:
This fast fail silently enforces a design decision that is elsewhere. If the conversion is possible and correct within the scope of this method (e.g. moving this to as_type), we should allow it.